addon/components/frost-list-pagination.js

```diff
@@ -77,14 +81,50 @@ export default Component.extend({
   @computed('_offset', '_end', 'total')
   _paginationText (_offset, _end, total) {
     return total === 0 ? '0 results found' : `${_offset} to ${_end} of ${total}`
-  }
-
+  },
+  @readOnly
+  @computed('total', 'itemsPerPage')
+  _lastPage (total, itemsPerPage) {
+    return Math.floor((total - 1) / itemsPerPage)
+  },
   // == Functions =============================================================
-
+  _sendDebounceOnChange (page) {
+    this.sendAction('onChange', page)
+    this.set('pageClickCounter', 0)
+  },
+  /* eslint-disable complexity */
+  _handleDebouncedOnChange (page, debounceInterval) {
+    const currentPage = this.get('page')
+
+    if (Math.abs(currentPage - page) > 1) {
+      const debouncedCall = this.get('_debouncedPaging')
+      run.cancel(debouncedCall)
+      this._sendDebounceOnChange(page)
+    } else {
+      const totalPages = this.get('_lastPage')
+      let counter = this.get('pageClickCounter') || 0
```

Comment:
Use `this.getWithDefault('pageClickCounter', 0)` instead.

Line 36 has set `_pageClickCounter` to `0` but then it is never used in the code anywhere.  I'm guessing that it should have not had an underscore and then it's value would have been what was expected here and having to default it to `0` wouldn't even be necessary?